app/src/main/java/com/invisibleteam/goinvisible/mvvm/settings/SettingsActivity.java

```diff
@@ -95,23 +96,43 @@ private void openIntervalsDialog() {
         alertDialog.show();
     }
 
-    private AlertDialog createIntervalsDialog() {
-        AlertDialog.Builder alertDialog = new AlertDialog.Builder(this);
-        alertDialog.setTitle(getString(R.string.intervals_dialog_title));
-
+    private Dialog createIntervalsDialog() {
```

Comment:
I think there is too much going on in this method. I would like you to split this method in multiple methods, or extract some code to separate classes.